cmd/node/factory/structs.go

```diff
@@ -1304,6 +1358,8 @@ func newBlockProcessorAndTracker(
 func newShardBlockProcessorAndTracker(
```

Comment:
There's an unneeded `nodesConfig *sharding.NodesSetup` parameter in this function, if is not needed anymore it can be removed

Reply:
I have noticed it, but I would rather do this in a separate pull request as there are lots of propagating changes.